--- original/Modules/_decimal/libmpdec/mpdecimal.h	2022-09-06 09:11:33.000000000 +0200
+++ patched/Modules/_decimal/libmpdec/mpdecimal.h	2022-09-12 19:19:48.313667083 +0200
@@ -135,6 +135,9 @@
   #elif defined(__x86_64__)
     #define CONFIG_64
     #define ASM
+  #elif defined(__arm64__)
+    #define CONFIG_64
+    #define ANSI
   #else
     #error "unknown architecture for universal build."
   #endif
