--- Modules/Setup	2022-09-05 15:00:02.000000000 +0200
+++ Modules/Setup.patched	2022-09-12 16:37:44.306831222 +0200
@@ -204,17 +204,17 @@
 #_csv _csv.c
 
 # Socket module helper for socket(2)
-#_socket socketmodule.c
+_socket socketmodule.c
 
 # Socket module helper for SSL support; you must comment out the other
 # socket line above, and edit the OPENSSL variable:
-# OPENSSL=/path/to/openssl/directory
-# _ssl _ssl.c \
-#     -I$(OPENSSL)/include -L$(OPENSSL)/lib \
-#     -lssl -lcrypto
-#_hashlib _hashopenssl.c \
-#     -I$(OPENSSL)/include -L$(OPENSSL)/lib \
-#     -lcrypto
+OPENSSL=__openssl_install_dir__
+_ssl _ssl.c \
+    -I$(OPENSSL)/include -L$(OPENSSL)/lib \
+    -lssl -lcrypto
+_hashlib _hashopenssl.c \
+    -I$(OPENSSL)/include -L$(OPENSSL)/lib \
+    -lcrypto
 
 # To statically link OpenSSL:
 # _ssl _ssl.c \
