--- original/Modules/_decimal/libmpdec/mpdecimal.h	2021-09-04 05:49:41.000000000 +0200
+++ patched/Modules/_decimal/libmpdec/mpdecimal.h	2022-09-12 20:21:43.738529552 +0200
@@ -135,6 +135,9 @@
   #elif defined(__x86_64__)
     #define CONFIG_64
     #define ASM
+  #elif defined(__arm64__)
+    #define CONFIG_64
+    #define ANSI
   #else
     #error "unknown architecture for universal build."
   #endif
