--- original/Mac/Tools/pythonw.c	2020-04-19 23:13:39.000000000 +0200
+++ patched/Mac/Tools/pythonw.c	2022-09-12 18:23:30.242681069 +0200
@@ -116,10 +116,16 @@
 
 #elif defined(__ppc__)
     cpu_types[0] = CPU_TYPE_POWERPC;
+
 #elif defined(__i386__)
     cpu_types[0] = CPU_TYPE_X86;
+
+#elif defined(__arm64__)
+    cpu_types[0] = CPU_TYPE_ARM64;
+
 #else
 #       error "Unknown CPU"
+
 #endif
 
     if (posix_spawnattr_setbinpref_np(spawnattr, count,
