From 5085f81493ad08b6a1141e459aca58907c661c94 Mon Sep 17 00:00:00 2001
From: Daniel Fritsi <ffddani@gmail.com>
Date: Sat, 19 Aug 2023 18:30:52 +0200
Subject: [PATCH] Fix arm64e related compilation issue

---
 unix/configure | 22 +++++++++++-----------
 unix/tcl.m4    | 14 +++++++-------
 2 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/unix/configure b/unix/configure
index 57d5081612..65054deb24 100755
--- a/unix/configure
+++ b/unix/configure
@@ -7669,14 +7669,14 @@ echo "${ECHO_T}$tcl_cv_cc_arch_x86_64" >&6
 fi
 ;;
 		    arm64|arm64e)
-			echo "$as_me:$LINENO: checking if compiler accepts -arch arm64e flag" >&5
-echo $ECHO_N "checking if compiler accepts -arch arm64e flag... $ECHO_C" >&6
-if test "${tcl_cv_cc_arch_arm64e+set}" = set; then
+			echo "$as_me:$LINENO: checking if compiler accepts -arch arm64 flag" >&5
+echo $ECHO_N "checking if compiler accepts -arch arm64 flag... $ECHO_C" >&6
+if test "${tcl_cv_cc_arch_arm64+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
 			    hold_cflags=$CFLAGS
-			    CFLAGS="$CFLAGS -arch arm64e"
+			    CFLAGS="$CFLAGS -arch arm64"
 			    cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -7714,22 +7714,22 @@ if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  tcl_cv_cc_arch_arm64e=yes
+  tcl_cv_cc_arch_arm64=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-tcl_cv_cc_arch_arm64e=no
+tcl_cv_cc_arch_arm64=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 			    CFLAGS=$hold_cflags
 fi
-echo "$as_me:$LINENO: result: $tcl_cv_cc_arch_arm64e" >&5
-echo "${ECHO_T}$tcl_cv_cc_arch_arm64e" >&6
-			if test $tcl_cv_cc_arch_arm64e = yes; then
+echo "$as_me:$LINENO: result: $tcl_cv_cc_arch_arm64" >&5
+echo "${ECHO_T}$tcl_cv_cc_arch_arm64" >&6
+			if test $tcl_cv_cc_arch_arm64 = yes; then
 
-			    CFLAGS="$CFLAGS -arch arm64e"
+			    CFLAGS="$CFLAGS -arch arm64"
 			    do64bit_ok=yes
 
 fi
@@ -7742,7 +7742,7 @@ echo "$as_me: WARNING: Don't know how enable 64-bit on architecture \`arch\`" >&
 else
 
 		# Check for combined 32-bit and 64-bit fat build
-		if echo "$CFLAGS " |grep -E -q -- '-arch (ppc64|x86_64|arm64e) ' \
+		if echo "$CFLAGS " |grep -E -q -- '-arch (ppc64|x86_64|arm64) ' \
 		    && echo "$CFLAGS " |grep -E -q -- '-arch (ppc|i386) '; then
 
 		    fat_32_64=yes
diff --git a/unix/tcl.m4 b/unix/tcl.m4
index ca94abd4ad..2f0f2c32bc 100644
--- a/unix/tcl.m4
+++ b/unix/tcl.m4
@@ -1542,15 +1542,15 @@ AC_DEFUN([SC_CONFIG_CFLAGS], [
 			    do64bit_ok=yes
 			]);;
 		    arm64|arm64e)
-			AC_CACHE_CHECK([if compiler accepts -arch arm64e flag],
-				tcl_cv_cc_arch_arm64e, [
+			AC_CACHE_CHECK([if compiler accepts -arch arm64 flag],
+				tcl_cv_cc_arch_arm64, [
 			    hold_cflags=$CFLAGS
-			    CFLAGS="$CFLAGS -arch arm64e"
+			    CFLAGS="$CFLAGS -arch arm64"
 			    AC_LINK_IFELSE([AC_LANG_PROGRAM([[]], [[]])],
-				    [tcl_cv_cc_arch_arm64e=yes],[tcl_cv_cc_arch_arm64e=no])
+				    [tcl_cv_cc_arch_arm64=yes],[tcl_cv_cc_arch_arm64=no])
 			    CFLAGS=$hold_cflags])
-			AS_IF([test $tcl_cv_cc_arch_arm64e = yes], [
-			    CFLAGS="$CFLAGS -arch arm64e"
+			AS_IF([test $tcl_cv_cc_arch_arm64 = yes], [
+			    CFLAGS="$CFLAGS -arch arm64"
 			    do64bit_ok=yes
 			]);;
 		    *)
@@ -1558,7 +1558,7 @@ AC_DEFUN([SC_CONFIG_CFLAGS], [
 		esac
 	    ], [
 		# Check for combined 32-bit and 64-bit fat build
-		AS_IF([echo "$CFLAGS " |grep -E -q -- '-arch (ppc64|x86_64|arm64e) ' \
+		AS_IF([echo "$CFLAGS " |grep -E -q -- '-arch (ppc64|x86_64|arm64) ' \
 		    && echo "$CFLAGS " |grep -E -q -- '-arch (ppc|i386) '], [
 		    fat_32_64=yes])
 	    ])
-- 
2.41.0
